spec/controllers/search_spec.rb

```diff
@@ -34,43 +35,33 @@
         expect(Gif.count).not_to eq(gifs_count)
       end
 
-      it "should count votes when a gif is upvoted or downvoted" do
-        post :slack, post_params
-
-        gif = Lastgif.find(team_domain: team.domain, channel: "testing_reactif").first
-        votes = team.teamgifs.find_by_gif_id(gif.gif_id).votes
-
-        post_params[:text], post_params[:user_name] = "upvote", "user1"
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).not_to eq(votes)
-
-        post_params[:text], post_params[:user_name], = "downvote", "user2"
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).to eq(votes)
-      end
-
-      it "should not allow users to vote more than once a week the same gif" do
-
-        post :slack, post_params
-        gif = Lastgif.find(team_domain: team.domain, channel: "testing_reactif").first
-        votes = team.teamgifs.find_by_gif_id(gif.gif_id).votes
-
-        post_params[:text], post_params[:user_name] = "upvote", "user1"
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).to eq(votes + 1)
-
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).to eq(votes + 1)
-
-        post_params[:text] = "downvote"
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).to eq(votes + 1)
-
-        Timecop.freeze(Time.now + 8.days)
-        post :slack, post_params
-        expect(team.teamgifs.find_by_gif_id(gif.gif_id).votes).to eq(votes)
+      context "when voting on the last gif" do
+        before { post :slack, post_params }
+        let(:last_gif) { Lastgif.find(team_domain: team.domain, channel: channel_name).first }
+        let(:teamgif) { team.teamgifs.find_by_gif_id(last_gif.gif_id) }
+
+        it "upvoting should increment votes by 1" do
+          post_params[:text] = "upvote"
+          expect{post :slack, post_params}.to change{teamgif.reload.votes}.by(1)
+        end
+
+        it "downvoting should increment votes by -1" do
+          post_params[:text] = "downvote"
+          expect{post :slack, post_params}.to change{teamgif.reload.votes}.by(-1)
+        end
+
+        it "the same user should not be able to vote more than once a week on the same gif" do
+          post_params[:text] = "upvote"
+          expect{post :slack, post_params}.to change{teamgif.reload.votes}.by(1)
+          expect{post :slack, post_params}.not_to change{teamgif.reload.votes}
+          post_params[:text] = "downvote"
+          expect{post :slack, post_params}.not_to change{teamgif.reload.votes}
+          Timecop.freeze(Time.now + 8.days)
+          expect{post :slack, post_params}.to change{teamgif.reload.votes}.by(-1)
```

Comment:
Line is too long. [81/80]<br>Space missing to the left of {.<br>Space missing inside {.<br>Space missing inside }.